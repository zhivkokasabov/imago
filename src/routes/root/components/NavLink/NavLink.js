import React from 'react';
import { Link } from 'react-router-dom';

import styles from './NavLink.module.scss';

import classNames from 'classnames';

export default function NavLink({
  href,
  text,
  className
}) {
  return <Link
    className={classNames(className, styles.link)}
    to={href}
  >
    {text}
  </Link>;
}