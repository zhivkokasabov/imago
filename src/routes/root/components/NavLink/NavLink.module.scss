.link {
  color: rgb(var(--text-default));
  font-size: .875rem;
  text-align: center;
  font-style: normal;
  font-weight: 400;
  line-height: 1.1875rem;
  text-decoration: none;

  &:focus, &:visited, &:active {
    color: rgb(var(--text-default));
  }
}