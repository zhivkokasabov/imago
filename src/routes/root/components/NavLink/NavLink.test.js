import React from 'react';
import { render } from '@testing-library/react';
import {
  MemoryRouter,
} from 'react-router-dom';

import NavLink from './NavLink';

describe('NavLink component', () => {
  it('should render NavLink', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavLink text="nav link" href="/" />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
