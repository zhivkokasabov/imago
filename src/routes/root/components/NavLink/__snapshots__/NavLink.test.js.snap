// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component should render NavLink 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="/"
  >
    nav link
  </a>
</DocumentFragment>
`;
